### PR TITLE
fix HMR for streaming responses

### PR DESCRIPTION
**Pull request by sokra**
### Description

this adds HMR support for streaming responses

### CODE REVIEW

File: crates/turbopack-dev-server/src/update/server.rs [hunk 1]
```diff
@@ -72,8 +72,17 @@ impl<P: SourceProvider + Clone + Send + Sync> UpdateServer<P> {
...
                                 }
                             };
-                            let stream = UpdateStream::new(TransientInstance::new(Box::new(get_content))).await?;
-                            streams.insert(resource, stream);
+                            match UpdateStream::new(TransientInstance::new(Box::new(get_content))).await {
+                                Ok(stream) => {
+                                    streams.insert(resource, stream);
+                                }
+                                Err(err) => {
+                                    println!("Failed to create update stream for {resource}: {}", PrettyPrintError(&err));
```
Comment (alexkirsz) [inline crates/turbopack-dev-server/src/update/server.rs:80]:
```suggestion
                                    eprintln!("Failed to create update stream for {resource}: {}", PrettyPrintError(&err));
```

File: crates/turbopack-core/src/version.rs [hunk 2]
```diff
@@ -186,17 +185,20 @@ pub enum Update {
 #[derive(PartialEq, Eq, Debug, Clone, TraceRawVcs, ValueDebugFormat, Serialize, Deserialize)]
 pub struct TotalUpdate {
     /// The version this update will bring the object to.
-    pub to: VersionVc,
+    #[turbo_tasks(trace_ignore)]
+    pub to: TraitRef<VersionVc>,
```
Comment (alexkirsz) [inline crates/turbopack-core/src/version.rs:189]: Why is this necessary?

Comment (sokra) [reply crates/turbopack-core/src/version.rs:189]: `compute_update_stream` is only executed the `UpdateStreamItem` changes (since it only this is awaited). But `VersionVc` might be equal while the underlying content can change. So when trying to send 2 `TotalUpdate` with `to` pointing to the same Vc (but with different content) it won't send that.

**Review by alexkirsz — APPROVED** [review]
